### PR TITLE
Move PCKMethodRedirector to Foundation+SpecHelper and make it public

### PR DESCRIPTION
**Pull request by younata**

### CODE REVIEW

**Comment by tjarratt** [comment]
*VERY* Interesting idea!

What's the use case? I can see this being helpful (I've certainly wanted to use it on projects...), but there's also probably the counter argument of "Method redirection and other runtime shenanigans are dangerous, and best applied sparingly. Making it easy is a double-edged sword".

Also, did you know [there is already a spec for this](https://github.com/pivotal/PivotalCoreKit/blob/master/Foundation/Spec/Extensions/NSObjectSpec+MethodRedirection.mm)?

The spec you added seems pretty small, but I can't tell if it adds anything new. Perhaps we should just consolidate them under a unified name?

**Comment by tjarratt** [comment]
OH wow this is from February. 

Where did the time go?

That was such a simpler time...

**Comment by younata** [comment]
Of course there's a not-at-all similarly named spec for this. :/

Use case is when I want to make my own custom stubs. rNews makes use of this behavior in a few places. ([WKWebView stubs](https://github.com/younata/RSSClient/blob/master/RSSClientTests/Fakes/WKWebView%2BrNewsTests.m), [UIViewController showController/showDetailController stubs](https://github.com/younata/RSSClient/blob/master/RSSClientTests/Fakes/UIViewController%2BrNewsTests.m), [UITraitCollection stubs](https://github.com/younata/RSSClient/blob/master/RSSClientTests/Fakes/UITraitCollection%2BrNewsTests.m)).

Basically, places where I wanted to make a stub, but didn't care to take the time to contribute the stub to PCK.

Relatedly, I should take some time and make pull requests for those...

**Comment by younata** [comment]
Huh. There is apparently also an NSObject+MethodDecorator category that... I don't really see the point of having.

Might open up another pull request to delete it, unless you have other thoughts?

@tjarratt @akitchen

**Comment by akitchen** [comment]
Maybe start by removing that compilation unit and running all the specs?

I believe it is there to allow renaming of existing methods in order to keep them available via a category interface, I.e -[SomeClass foo_orig:]

**Comment by tjarratt** [comment]
`+decorateMethod:(NSString *)methodName with:(NSString *)decoratedName` does appear to be entirely unused (within PCK). Seems like deleting it is eminently reasonable, from the perspective of maintaining the internals of PCK.

However, I think (think?) it is technically part of the public API, [since it's present in what appears to be a public header](https://github.com/pivotal/PivotalCoreKit/search?utf8=%E2%9C%93&q=MethodDecoration.h+&type=Code).

Might be worth keeping it around for this PR and then tackle it in a separate PR.

**Comment by akitchen** [comment]
I think this may have been left over from the old NSURLConnection spec helper code. I agree it's probably safe to delete if there are feelings that it's no longer needed.
